src/OpenLoco/Vehicles/VehicleHead.cpp

```diff
@@ -2676,9 +2676,18 @@ namespace OpenLoco::Vehicles
 
                 break;
             }
+
+            const bool stationHadPreviousCargo = cargoStats.quantity != 0;
             cargoStats.quantity = Math::Bound::add(cargoStats.quantity, cargo.qty);
             station->updateCargoDistribution();
-            cargoStats.enroute_age = std::max(cargoStats.enroute_age, cargo.numDays);
+            cargoStats.enrouteAge = Math::Bound::add(cargoStats.enrouteAge, cargo.numDays);
+            if (stationHadPreviousCargo)
+            {
+                // enrouteAge = enrouteAge * (1 - addedQuantity / summedQuantity)
```

Comment:
Add a comment that this is calculating the weighted average of the two cargo sources and that this is a change from vanilla.